verification/verify_test.go

```diff
@@ -85,27 +89,98 @@ func TestVerify(t *testing.T) {
 	assert.Nil(res)
 	assert.Equal(ErrPixelsAbsent, err)
 
+	// Check pixel count succeeds w/o external verifier
+	sv = &SegmentVerifier{policy: &Policy{Verifier: nil, Retries: 2}, verifySig: func(ethcommon.Address, []byte, []byte) bool { return true }}
+	pxls, err := pixels("../server/test.flv")
+	a := make([]byte, pxls)
+	assert.Nil(err)
+	data = &net.TranscodeData{Segments: []*net.TranscodedSegmentData{
+		{Url: "../server/test.flv", Pixels: pxls},
+		{Url: "../server/test.flv", Pixels: pxls},
+	}}
+	renditions := [][]byte{a, a}
+	res, err = sv.Verify(&Params{Results: data, Orchestrator: &net.OrchestratorInfo{TicketParams: &net.TicketParams{}}, Renditions: renditions})
+	assert.Nil(err)
+	assert.NotNil(res)
+
 	// check pixel count fails w/o external verifier w populated renditions but incorrect pixel counts
-	pixels, err := pixels("../server/test.flv")
+	sv = NewSegmentVerifier(&Policy{Verifier: nil, Retries: 2}) // reset
+	pxls, err = pixels("../server/test.flv")
 	assert.Nil(err)
 	data = &net.TranscodeData{Segments: []*net.TranscodedSegmentData{
 		{Url: "../server/test.flv", Pixels: 123},
 		{Url: "../server/test.flv", Pixels: 456},
 	}}
-	renditions := [][]byte{{byte(pixels)}, {byte(pixels)}}
+	renditions = [][]byte{{byte(pxls)}, {byte(pxls)}}
 	res, err = sv.Verify(&Params{Results: data, Orchestrator: &net.OrchestratorInfo{}, Renditions: renditions})
 	assert.Nil(res)
 	assert.Equal(ErrPixelMismatch, err)
 
 	// Check pixel count succeeds w/o external verifier
 	data = &net.TranscodeData{Segments: []*net.TranscodedSegmentData{
-		{Url: "../server/test.flv", Pixels: pixels},
-		{Url: "../server/test.flv", Pixels: pixels},
+		{Url: "../server/test.flv", Pixels: pxls},
+		{Url: "../server/test.flv", Pixels: pxls},
 	}}
 	res, err = sv.Verify(&Params{Results: data, Orchestrator: &net.OrchestratorInfo{}, Renditions: renditions})
 	assert.Nil(err)
 	assert.NotNil(res)
 
+	// Check sig verifier fails when len(params.Results.Segments) != len(params.Renditions)
+	pxls, err = pixels("../server/test.flv")
+	assert.Nil(err)
+	data = &net.TranscodeData{Segments: []*net.TranscodedSegmentData{
+		{Url: "../server/test.flv", Pixels: pxls},
+		{Url: "../server/test.flv", Pixels: pxls},
+	}}
+	renditions = [][]byte{}
+	res, err = sv.Verify(&Params{Results: data, Orchestrator: &net.OrchestratorInfo{TicketParams: &net.TicketParams{}}, Renditions: renditions})
+	assert.Equal(errPMCheckFailed, err)
+	assert.Nil(res)
+
+	// Check sig verifier passes when len(params.Results.Segments) == len(params.Renditions) and sig is valid
+	// We use this stubVerifier to make sure that ffmpeg doesnt try to read from a file
+	sv = NewSegmentVerifier(&Policy{Verifier: &stubVerifier{
+		results: nil,
+		err:     nil,
+	}, Retries: 2})
+	testData0 := []byte{1, 2, 3, 4, 5, 6, 7, 8}
```

Comment:
I don't think we need this test data setup now that we can stub the `verifySig` field in `SegmentVerifier` with an implementation that returns true/false.

Reply:
We still need the test data setup in order for the code to generate a proper signature.   If if we don't need it, is there any reason to remove it?  Please keep in mind that tests also serve as documentation for developers.

Reply:
understood.

done.